packages/block-editor/src/components/block-settings-menu-controls/index.js

```diff
@@ -70,8 +78,17 @@ const BlockSettingsMenuControlsSlot = ( { fillProps, clientIds = null } ) => {
 	const convertToGroupButtonProps =
 		useConvertToGroupButtonProps( selectedClientIds );
 	const { isGroupable, isUngroupable } = convertToGroupButtonProps;
+
+	// Don't show ungroup for section blocks when the experiment is enabled
+	// since we show "Unlock design" instead
+	const shouldShowUngroup =
+		isUngroupable &&
+		! (
+			isSectionBlock && window?.__experimentalContentOnlyPatternInsertion
+		);
+
```

Comment:
Once we merge #73183 we won't need this. It's simpler to ensure that section blocks are never ungroupable, because there aren't any circumstances under which we'd want to ungroup them.

Reply:
Yeah this all got removed in the rebase.